src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySourceOptions.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    /// <summary>
+    /// Options for creating a <see cref="ActivitySource"/>.
+    /// </summary>
+    public class ActivitySourceOptions
+    {
+        private string _name;
+
+        /// <summary>
+        /// Constructs a new instance of <see cref="ActivitySourceOptions"/>.
+        /// </summary>
+        /// <param name="name">The name of the <see cref="ActivitySourceOptions"/> object</param>
+        public ActivitySourceOptions(string name)
+        {
+            _name = name ?? throw new ArgumentNullException(nameof(name));
+        }
+
+        /// <summary>
+        /// Get or set the <see cref="ActivitySourceOptions"/> object name. Cannot be null.
+        /// </summary>
+        public string Name
+        {
+            get => _name;
+            set
+            {
+                if (value is null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                _name = value;
+            }
+        }
+
+        /// <summary>
+        /// The optional <see cref="ActivitySourceOptions"/> version. Defaulted to empty string.
+        /// </summary>
+        public string? Version { get; set; } = string.Empty;
```

Comment:
It defaults to empty but it's allowed to be null?

Reply:
Yes, the version can be set to null.

Reply:
No. Both values indicate the version is not set.